### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by OnlyUnknown**
- Have added nameable class and its requirements 
- Have added Decorator class and its requirements
- Have added Capitalize and Trim inherited to Decorator and have added the requirement

### CODE REVIEW

File: Ruby-Project/person_class.rb [hunk 1]
```diff
@@ -23,4 +25,31 @@ def of_age?
 
     end
   end
+
+  def correct_name
+    @name
+  end
```
Comment (julie-ify) [inline Ruby-Project/person_class.rb]: - [ ] This should not be a private method. Kindly remove it from under the `private` keyword.

- [ ] Kindly fix these broken links in your readme headers. You can remove those sections if you have no sections intended for them.

  image

File: Ruby-Project/person_class.rb [hunk 2]
```diff
@@ -23,4 +25,31 @@ def of_age?
...
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0..9]
+  end
```
Comment (julie-ify) [inline Ruby-Project/person_class.rb]: - [ ] **[OPTIONAL]** These classes should be in their own files. Each class is usually located in its own file, according to Ruby conventions.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @OnlyUnknown,

Good job so far!

## Highlights!
- Good code organization and decorator classes ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**